File: WebLeave/SPA/src/app/_core/models/common/leave-data.ts
import { LunchBreak } from "./lunch-break";

export interface LeaveData {
  leaveID: number;
  leaveArchive: string;
  empID: number | null;
  cateID: number | null;
  time_Start: string | null;
  time_End: string | null;
  dateLeave: string | null;
  leaveDay: number | null;
  approved: number | null;
  time_Applied: string | null;
  timeLine: string;
  comment: string;
  leavePlus: boolean | null;
  leaveArrange: boolean | null;
  userID: number | null;
  approvedBy: number | null;
  editRequest: number | null;
  status_Line: boolean | null;
  created: string | null;
  updated: string | null;
  status_Lock: boolean | null;
  mailContent_Lock: string;
  commentArchive: string;
  reasonAdjust: string;

  empName: string | null;
  empNumber: string | null;
  partCode: string | null;
  partCodeTruncate: string | null;
  cateSym: string | null;
  cateNameVN: string | null;
  cateNameEN: string | null;
  cateNameZH: string | null;
  cateName_vi: string | null;
  cateName_en: string | null;
  cateName_zh: string | null;
  leaveDayByString: string | null;
  checkbox: boolean | null;
  exhibit: boolean | null;
  partID: number | null;
  partNameVN: string | null;
  partNameEN: string | null;
  partNameZH: string | null;
  partName_vi: string | null;
  partName_en: string | null;
  partName_zh: string | null;
  deptNameVN: string | null;
  deptNameEN: string | null;
  deptNameZH: string | null;
  deptName_vi: string | null;
  deptName_en: string | null;
  deptName_zh: string | null;
  dateIn: string | null;
  commentLeave: number | null;
  deptCode: string;
  approvedString: string;
  sender: string | null;
  sendBy: string | null;
  translatedComment: string | null;

  numberDay: string;
  category: string;
  statusString: string;
  fullName: string;
  gBID: number | null;
  partSym: string;
  deptSym: string;
  buildingSym: string;
  areaSym: string;
  approveName: string;
  roleEditComments: number;
  employeeId: string;
  commentList: string[];
  departmentName: string;
  approveList: string[];
  lunchBreakVN: string;
  lunchBreakEN: string;
  lunchBreakZH: string;
}

export interface LeaveDetail {
  editCommentArchive: boolean;
  notiUser: boolean;
  roleApproved: boolean;
  enablePreviousMonthEditRequest: boolean;
  leaveData: LeaveData;
  approvalPersons: string[];
  deletedLeaves: LeaveDeleteDetail[];
}

export interface LeaveDeleteDetail {
  timeLine: string;
  comment: string;
  cateNameVN: string;
  cateNameEN: string;
  cateNameZH: string;
  translatedComment: string;
}

File: WebLeave/API/Models/LeaveData.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace API.Models
{
    public partial class LeaveData
    {
        [Key]
        public int LeaveID { get; set; }
        [StringLength(30)]
        [Unicode(false)]
        public string LeaveArchive { get; set; }
        public int? EmpID { get; set; }
        public int? CateID { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? Time_Start { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? Time_End { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? DateLeave { get; set; }
        public double? LeaveDay { get; set; }
        public int? Approved { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? Time_Applied { get; set; }
        [StringLength(50)]
        [Unicode(false)]
        public string TimeLine { get; set; }
        public string Comment { get; set; }
        public bool? LeavePlus { get; set; }
        public bool? LeaveArrange { get; set; }
        public int? UserID { get; set; }
        public int? ApprovedBy { get; set; }
        public int? EditRequest { get; set; }
        public bool? Status_Line { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? Created { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? Updated { get; set; }
        public bool? Status_Lock { get; set; }
        [StringLength(255)]
        public string MailContent_Lock { get; set; }
        [StringLength(500)]
        public string CommentArchive { get; set; }
        [StringLength(500)]
        public string ReasonAdjust { get; set; }

        [ForeignKey("ApprovedBy")]
        [InverseProperty("LeaveDataApprovedByNavigations")]
        public virtual Users ApprovedByNavigation { get; set; }
        [ForeignKey("CateID")]
        [InverseProperty("LeaveData")]
        public virtual Category Cate { get; set; }
        [ForeignKey("EmpID")]
        [InverseProperty("LeaveData")]
        public virtual Employee Emp { get; set; }
        [ForeignKey("UserID")]
        [InverseProperty("LeaveDataUsers")]
        public virtual Users User { get; set; }
    }
}